### PR TITLE
refactor: Remove LoggerWrapper in favor of Logger refs

### PR DESCRIPTION
**Pull request by paulgessinger**
To harmonize / simplify the ways that loggers are used, this PR removes `LoggerWrapper`. To support the case of e.g. functions optionally accepting a logger, we can use `const Logger&` and default it to `getDummyLogger`. Since `getDummyLogger` returns a reference to a logger instance of static lifetime, all the arguments should be safe, even if they are retained (I think).

### CODE REVIEW

File: Alignment/include/ActsAlignment/Kernel/Alignment.hpp [hunk 1]
```diff
@@ -142,7 +136,10 @@ struct Alignment {
   Alignment() = delete;
 
   /// Constructor from arguments
-  Alignment(fitter_t fitter) : m_fitter(std::move(fitter)) {}
+  Alignment(fitter_t fitter,
+            std::unique_ptr<const Acts::Logger> _logger =
+                Acts::getDefaultLogger("Alignment", Acts::Logging::INFO))
+      : m_fitter(std::move(fitter)), m_logger{std::move(_logger)} {}
```
Comment (andiwand) [inline Alignment/include/ActsAlignment/Kernel/Alignment.hpp:142]:
```suggestion
            std::unique_ptr<const Acts::Logger> logger =
                Acts::getDefaultLogger("Alignment", Acts::Logging::INFO))
      : m_fitter(std::move(fitter)), m_logger{std::move(logger)} {}
```

File: Core/include/Acts/EventData/detail/CorrectedTransformationFreeToBound.hpp [hunk 2]
```diff
@@ -91,7 +91,7 @@ struct CorrectedFreeToBoundTransformer {
       const FreeVector& freeParams, const FreeSymMatrix& freeCovariance,
       const Surface& surface, const GeometryContext& geoContext,
       NavigationDirection navDir = NavigationDirection::Forward,
-      LoggerWrapper logger = getDummyLogger()) const;
+      const Logger& logger = getDummyLogger()) const;
```
Comment (andiwand) [inline Core/include/Acts/EventData/detail/CorrectedTransformationFreeToBound.hpp:94]: I guess in the best case we never end up with a dummy logger but it gets always passed down from the user, right?

File: Core/include/Acts/Utilities/Intersection.hpp [hunk 3]
```diff
@@ -198,22 +198,17 @@ namespace detail {
...
 
   const bool coCriterion = cLimit > oLimit;
   const bool cpCriterion = std::abs(cLimit) < std::abs(pLimit) + tolerance;
 
   const bool accept = coCriterion and cpCriterion;
 
-  if constexpr (doLogging) {
     if (accept) {
       ACTS_VERBOSE("Intersection is WITHIN limit");
     } else {
```
Comment (andiwand) [inline Core/include/Acts/Utilities/Intersection.hpp:214]: I guess this and the following should be moved to the left?

File: Examples/Algorithms/TrackFitting/src/TrackFittingAlgorithm.cpp [hunk 4]
```diff
@@ -74,13 +74,9 @@ ActsExamples::ProcessCode ActsExamples::TrackFittingAlgorithm::execute(
...
-  GeneralFitterOptions options{ctx.geoContext,
-                               ctx.magFieldContext,
-                               ctx.calibContext,
-                               calibrator,
-                               &(*pSurface),
-                               Acts::LoggerWrapper{logger()},
-                               Acts::PropagatorPlainOptions()};
+  GeneralFitterOptions options{
+      ctx.geoContext, ctx.magFieldContext, ctx.calibContext,
+      calibrator,     &(*pSurface),        Acts::PropagatorPlainOptions()};
```
Comment (andiwand) [inline Examples/Algorithms/TrackFitting/src/TrackFittingAlgorithm.cpp:79]: I think this needs to be reformatted

File: Plugins/DD4hep/include/Acts/Plugins/DD4hep/ConvertDD4hepMaterial.hpp [hunk 5]
```diff
@@ -27,10 +27,10 @@ class Layer;
 /// @param detElement the DD4hep detector element for which this material is
 ///                   assigned
 /// @param cylinderLayer is the target layer
-/// @param logger a @c LoggerWrapper for output
+/// @param logger a @c Logger  for output
```
Comment (andiwand) [inline Plugins/DD4hep/include/Acts/Plugins/DD4hep/ConvertDD4hepMaterial.hpp:30]:
```suggestion
/// @param logger a @c Logger for output
```

**Review by andiwand — APPROVED** [review]
looks good! thanks for this :+1:

File: Alignment/include/ActsAlignment/Kernel/Alignment.hpp (hunk 1 above)
Comment (paulgessinger) [reply Alignment/include/ActsAlignment/Kernel/Alignment.hpp:142]: I think this shadows the `const Logger& logger() const;` method, no?

File: Core/include/Acts/EventData/detail/CorrectedTransformationFreeToBound.hpp (hunk 2 above)
Comment (paulgessinger) [reply Core/include/Acts/EventData/detail/CorrectedTransformationFreeToBound.hpp:94]: That's what I'm thinking yeah. The dummy logger is a tradeoff where we default to not logging at all. That's preferable over constructing a local logger in the function body (which wasn't actually the case in this specific instance before this PR, anyway)

File: Core/include/Acts/Utilities/Intersection.hpp (hunk 3 above)
Comment (paulgessinger) [reply Core/include/Acts/Utilities/Intersection.hpp:214]: Huh, clang-tidy doesn't complain?

Comment (paulgessinger) [reply Core/include/Acts/Utilities/Intersection.hpp:214]: Nevermind, it does.

File: Examples/Algorithms/TrackFitting/src/TrackFittingAlgorithm.cpp (hunk 4 above)
Comment (paulgessinger) [reply Examples/Algorithms/TrackFitting/src/TrackFittingAlgorithm.cpp:79]: Seems like clang-format wants it this way, for some reason.

**Review by andiwand — APPROVED** [review]

**Review by andiwand — APPROVED** [review]

**Pull request merged**